packages/core/src/node/logger-cli-contribution.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2018 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as yargs from 'yargs';
+import { injectable } from 'inversify';
+import { LogLevel } from '../common/logger';
+import { CliContribution } from './cli';
+import * as fsextra from 'fs-extra';
+import * as nsfw from 'nsfw';
+import { Event, Emitter } from '../common/event';
+import * as path from 'path';
+
+/** Convert string to LogLevel, throw if invalid.  */
+function readLogLevelString(levelStr: string, errMessagePrefix: string): LogLevel {
+    const level = LogLevel.fromString(levelStr);
+
+    if (level === undefined) {
+        throw new Error(`${errMessagePrefix}: "${levelStr}".`);
+    }
+
+    return level;
+}
+
+/** Maps logger names to log levels.  */
+export interface LogLevels {
+    [key: string]: LogLevel,
+}
+
+/**
+ * Parses command line switches related to log levels, then watches the log
+ * levels file (if specified) for changes.  This is the source of truth for
+ * what the log level per logger should be.
+ */
+@injectable()
+export class LogLevelCliContribution implements CliContribution {
+    logLevels: LogLevels = {};
+
+    /**
+     * Log level to use for loggers not specified in `logLevels`.
+     */
+    defaultLogLevel: LogLevel = LogLevel.INFO;
+
+    logLevelsChangeEvent: Emitter<{}> = new Emitter<{}>();
```

Comment:
please take care to mark internals as protected

Reply:
btw you can use `void` and fire `undefined`

Reply:
Ok, I'll try that.